API.md

```diff
@@ -0,0 +1,1246 @@
+        Lets you define in the Ship the name of the parameters you need to check for. You can return parameters in the Promise resolve and reject methods, that will be passed to the view. This lets you display status and show buttons and more to the customer
+    -   `options.onAuthorize` **[Function][61]** A method returning a Promise, resolved when complete. Best used to save tokens and continue the sequence once saved.
+    -   `options.onLogin` **[Function][61]** A method returning a Promise, resolved when ready. Best used to process form parameters, and place them in `req.authParams` to be submitted to the Login sequence. Useful to add strategy-specific parameters, such as a portal ID for Hubspot for instance.
+    -   `options.Strategy` **[Function][61]** A Passport Strategy.
+    -   `options.views` **[Object][57]** Required, A hash of view files for the different screens: login, home, failure, success
+    -   `options.options` **[Object][57]** Hash passed to Passport to configure the OAuth Strategy. (See [Passport OAuth Configuration][75])
+
+**Examples**
+
+```javascript
+const { oAuthHandler } = require("hull/lib/utils");
+const { Strategy as HubspotStrategy } = require("passport-hubspot");
+
+const app = express();
+
+app.use(
+  '/auth',
+  oAuthHandler({
+    name: 'Hubspot',
+    tokenInUrl: true,
+    Strategy: HubspotStrategy,
+    options: {
+      clientID: 'xxxxxxxxx',
+      clientSecret: 'xxxxxxxxx', //Client Secret
+      scope: ['offline', 'contacts-rw', 'events-rw'] //App Scope
+    },
+    isSetup(req) {
+      if (!!req.query.reset) return Promise.reject();
+      const { token } = req.hull.ship.private_settings || {};
+      return !!token
+      ? Promise.resolve({ valid: true, total: 2 })
+      : Promise.reject({ valid: false, total: 0 });
+    },
+    onLogin: req => {
+      req.authParams = { ...req.body, ...req.query };
+      return req.hull.client.updateSettings({
+        portalId: req.authParams.portalId
+      });
+    },
+    onAuthorize: req => {
+      const { refreshToken, accessToken } = req.account || {};
+      return req.hull.client.updateSettings({
+        refresh_token: refreshToken,
+        token: accessToken
+      });
+    },
+    views: {
+      login: 'login.html',
+      home: 'home.html',
+      failure: 'failure.html',
+      success: 'success.html'
+    }
+  })
+);
+
+//each view will receive the following data:
+{
+  name: "The name passed as handler",
+  urls: {
+    login: '/auth/login',
+    success: '/auth/success',
+    failure: '/auth/failure',
+    home: '/auth/home',
+  },
+  ship: ship //The entire Ship instance's config
+}
+```
+
+Returns **[Function][61]** OAuth handler to use with expressjs
+
+### smartNotifierHandler
+
+`smartNotifierHandler` is a next generation `notifHandler` cooperating with our internal notification tool. It handles Backpressure, throttling and retries for you and lets you adapt to any external rate limiting pattern.
```

Comment:
If we're deprecating `notifHandler` we don't want to refer to it here I guess. And we can rename `internal notification tool` to `Kraken`.